### DISCUSSION TITLE
Can you add option to change download priority?

### DISCUSSION THREAD

**Discussion by Defog7277** [bibanon/tubeup#283] [Ideas]
I am new to github and am still lost in its interface.
A week ago I wrote a proposal in the wrong section and only now I noticed that there is a proper section dedicated to discussions and proposals to make tubeup a better program.

I would like to archive some small streamer channels. So far I've been doing this using yt-dlp and manually entering metadata on archive.org. I was very happy to see that there is a program like tubeup that automates everything.

I would like to propose an option that changes the download priority. If not by default then so that you can set this priority yourself.

Currently it looks like the highest priority is to download livechat. When the program finishes downloading the livechat it then starts downloading the livestream recording.

Is it possible to add an option to change this priority? That is, downloading the livestream first and after finishing the downloading in the second order livechat.

Why do I care about this? Many streamers delete the recording of their live or set it as private right after finishing the broadcast and then we lose the possibility of archiving.

Of course, the problem only applies to tubeup, because with yt-dlp you can set yourself what you want to download first. It would be nice if you added such an option in tubeup

**Comment by vxbinaca** [comment]
You're trying to download live video, yes? We don't control yt-dlp which does the downloading. They handle download order.

Furthermore,[ live-from-start is experimental](https://github.com/bibanon/tubeup/issues/186#issuecomment-1127698704) and per yt-dlps maintainer not suitable for archival purposes.

**Comment by vxbinaca** [comment]
Also we don't set download priority, yt-dlp handles that. @brandongalbraith I don't think we force livechat first on live videos, but then again as I said in my last reply live videos aren't reliably done by the YT extractor (or any extractor) so we discourage use.

But download order I think is done by yt-dlp, am I correct?

**Comment by Defog7277** [comment]
> You're trying to download live video, yes? 

Not exactly. I use yt-dlp to download ongoing livestreams. I wanted to use Tubeup to automate my archiving of these video files on archive.org

> We don't control yt-dlp which does the downloading. They handle download order.

Strange because when I use the `yt-dlp <livestream link>` command in youtube-dlp, it only downloads my video
If I wanted to download livechat then I would have to use the command `--sub-langs all --write-subs <livestreamlink>`
Of course I am referring to ongoing livestreams and not fully processed Vod's from livestreams

> Furthermore,https://github.com/bibanon/tubeup/issues/186#issuecomment-1127698704 and per yt-dlps maintainer not suitable for archival purposes.

For several months I have been downloading livestreams using the --live-from-start command. I think it will be several hundred video recordings. Only a few video files had problems, I think more than 95% of the files were OK. 

However, I fully understand that you are not going to release a version of Tubeup that is not fully trouble-free. 

But maybe there is a solution? What do you think about releasing an experimental or beta version that would have a default setting to download the video first?

**Comment by vxbinaca** [comment]
Brand does the order of the library calls in yt-dlp matter? This still sounds like a yt-dlp issue and not a us issue.

Edit: @pukkandan mind taking a look at this? Is this a us problem or something on your end?

**Reply by pukkandan** [reply]
Subtitles are always downloaded first.This is not a problem for normal videos, but for live streams, the live chat download takes till the end of stream to complete. Even if it were possible to reverse order of downloads, it wouldn't really help, since video will also take till end of stream and the live chat will instead fail to download in that case. This is not a bug per-se, but just yt-dlp's inability to perform multiple downloads in parallel.

The generally recommended workaround is to start two yt-dlp instances to do the two downloads. But I don't know if that is feasible for tubeup to implement. Failing that, it may be preferable to do `--sub-langs all,-live_chat` so that only video is downloaded and the live chat is skipped.

**Reply by vxbinaca** [reply]
thanks we'll make a wiki for Tubeup and I'll try to force deny live video rips as a way to discourage people from trying, wne we'll point this back to your two replies as to why we don't support live videos.

**Comment by vxbinaca** [comment]
Stop using tubeup to do your live video rips. For numerous reasons it's not technically possible to do uploads using our tool.

**Comment by brandongalbraith** [comment]
Late to the discussion, but here are my thoughts:

I am not opposed to improving the workflow around them, but the heavy lifting will be done by yt-dlp as the extractor. If the extraction experience is poor, head over to their repo and contribute improvements the the extractor in question. Once live rips are improved, we'll consider the effort to improve the tubeup experience around them. If you'd like to manually upload live rips, you can upload programmatically using https://github.com/jjjake/internetarchive post successful retrieval.

**Comment by vxbinaca** [comment]
@brandongalbraith Please see [this note](https://github.com/bibanon/tubeup/commit/8c6fd019f97e6cb3690a14a33bbcd46438ba071d) I attached to the readme
